dataset/table.py

```diff
@@ -12,67 +12,76 @@ class Table:
     """
     A wrapper class for Dataset Tables.
     """
-    def __init__(self, name, src, *args):
+    def __init__(self, name, src, na_values=None, exclude_attr_cols=['_tid_'],
+            fpath=None, df=None, db_conn=None, table_query=None, db_engine=None):
+        """
+        :param name: (str) name to assign to dataset.
+        :param na_values: (str or list[str]) values to interpret as NULL.
+        :param exclude_attrs: (list[str]) list of columns to NOT treat as
+            attributes during training/learning.
+        :param src: (Source) type of source to load from. Note additional
+            parameters MUST be provided for each specific source:
+                Source.FILE: :param fpath:, read from CSV file
+                Source.DF: :param df:, read from pandas DataFrame
+                Source.DB: :param db_conn:, read from database table with :param name:
+                Source.SQL: :param table_query: and :param db_engine:, use result
+                    from :param table_query:
+
+        :param fpath: (str)
+        :param df: (pandas.DataFrame)
+        :param db_conn: (SQLAlchemy connectable, str)
+        :param table_query: (str)
+        :param db_engine: (DBEngine)
+        """
         self.name = name
         self.index_count = 0
+        # Copy the list to memoize
+        self.exclude_attr_cols = list(exclude_attr_cols)
         self.df = pd.DataFrame()
+
         if src == Source.FILE:
-            if len(args) < 2:
-                raise Exception("ERROR while loading table. File path and file name expected.Please provide <file_path> and <file_name>.")
-            else:
-                file_path = args[0]
-                file_name = args[1]
-                if len(args) == 3:
-                    na_values = args[2]
-                else:
-                    na_values = None
-                # TODO(richardwu): use COPY FROM instead of loading this into memory
-                self.df = pd.read_csv(os.path.join(file_path,file_name), dtype=str, na_values=na_values)
-                # Normalize to lower strings and strip whitespaces.
-                # TODO: No support for numerical values. To be added.
-                for attr in self.df.columns.values:
-                    if attr != '_tid_' and self.df[attr].dtype == str:
-                        self.df[attr] = self.df[attr].apply(lambda x: x.lower())
-                        self.df[attr] = self.df[attr].apply(lambda x: x.strip())
-                self.df.columns = map(str.lower, self.df.columns)
+            if fpath is None:
+                raise Exception("ERROR while loading table. File path for CSV file name expected. Please provide <df> param.")
+            # TODO(richardwu): use COPY FROM instead of loading this into memory
```

Comment:
For the time being since we treat everything as a string (or object if there are nans) we could read the first line of the file to determine the column names which suffices for our schema. I think moving beyond string type attributes (e.g. real-valued attributes) is still a WIP research project.